### PR TITLE
chore: updating formatCurrencyAmount to handle multiple langs and cur…

### PR DESCRIPTION
**Pull request by JackShort**
* updates formatCurrencyAmount to accept locale and localCurrency
* also adds the formatting provided from the useFormatterLocals in #7267

_I am aware of the bug when typing in the swap panel in another formatted language that will be fixed in another pr_
_note: none of these effects are visible without enabling the feature flag_

Screen Shot 2023-09-05 at 3 59 34 PM

### CODE REVIEW

File: src/utils/formatNumbers.ts [hunk 1]
```diff
@@ -436,6 +436,30 @@ function useFormatterLocales(): {
...
 
+interface FormatCurrencyAmountOptions {
+  amount: Nullish<CurrencyAmount<Currency>>
+  type?: NumberType
+  placeholder?: string
+  locale?: SupportedLocale
+  localCurrency?: SupportedLocalCurrency
+}
+
+export function formatCurrencyAmount({
```
Comment (cbachmeier) [inline src/utils/formatNumbers.ts:447]: nit: could you add a description to this function and to `useFormatter` detailing what it does and how to use it?

**Review by cbachmeier — COMMENTED** [review]
Do we expect the amount to not be converted to the new currency? Only the symbol? In the screenshots below you can see my USD and YEN value is identical but yen should be 100x
Screenshot 2023-09-06 at 9 25 09 AM
Screenshot 2023-09-06 at 9 26 08 AM

**Comment by JackShort** [comment]
> Do we expect the amount to not be converted to the new currency? Only the symbol? In the screenshots below you can see my USD and YEN value is identical but yen should be 100x Screenshot 2023-09-06 at 9 25 09 AM Screenshot 2023-09-06 at 9 26 08 AM

yeah this doesn't actually apply the currency conversions yet - it only formats it in the other currency. going to handle the currency conversion in another pr lol

File: src/utils/formatNumbers.ts (hunk 1 above)
Comment (JackShort) [reply src/utils/formatNumbers.ts:447]: will do that in the base pr

File: src/components/AccountDrawer/MiniPortfolio/Activity/parseLocal.ts [hunk 2]
```diff
@@ -37,19 +38,25 @@ function buildCurrencyDescriptor(
...
 ) {
-  const formattedA = currencyA ? formatCurrencyAmount(CurrencyAmount.fromRawAmount(currencyA, amtA)) : t`Unknown`
+  const formattedA = currencyA
+    ? formatCurrencyAmount({ amount: CurrencyAmount.fromRawAmount(currencyA, amtA), locale })
+    : t`Unknown`
   const symbolA = currencyA?.symbol ?? ''
-  const formattedB = currencyB ? formatCurrencyAmount(CurrencyAmount.fromRawAmount(currencyB, amtB)) : t`Unknown`
+  const formattedB = currencyB
+    ? formatCurrencyAmount({ amount: CurrencyAmount.fromRawAmount(currencyB, amtB), locale })
+    : t`Unknown`
```
Comment (cbachmeier) [inline src/components/AccountDrawer/MiniPortfolio/Activity/parseLocal.ts:50]: this only seems to format new history that occurred with the flag on and not any past history. Is this due to caching? You can see this in the screenshot with the recent swap being comma separated and the old history still using a decimal
Screenshot 2023-09-06 at 9 33 36 AM

File: src/components/AccountDrawer/MiniPortfolio/Activity/parseLocal.ts [hunk 3]
```diff
@@ -168,7 +176,7 @@ export function transactionToActivity(
     let additionalFields: Partial<Activity> = {}
     const info = details.info
     if (info.type === TransactionType.SWAP) {
-      additionalFields = parseSwap(info, chainId, tokens)
+      additionalFields = parseSwap(info, chainId, tokens, locale)
```
Comment (cbachmeier) [inline src/components/AccountDrawer/MiniPortfolio/Activity/parseLocal.ts:179]: why not add locale to parseWrap and parseApproval as well?

File: src/components/Tokens/TokenDetails/BalanceSummary.tsx [hunk 4]
```diff
@@ -67,8 +67,15 @@ export default function BalanceSummary({ token }: { token: Currency }) {
   const theme = useTheme()
   const { label, color } = getChainInfo(asSupportedChain(chainId) ?? ChainId.MAINNET)
   const balance = useCurrencyBalance(account, token)
-  const formattedBalance = formatCurrencyAmount(balance, NumberType.TokenNonTx)
-  const formattedUsdValue = formatCurrencyAmount(useStablecoinValue(balance), NumberType.FiatTokenStats)
+  const { formatCurrencyAmount } = useFormatter()
```
Comment (cbachmeier) [inline src/components/Tokens/TokenDetails/BalanceSummary.tsx:70]: The Swap input panel bug with a formatted language also happens here as well
Screenshot 2023-09-06 at 9 39 09 AM
Screenshot 2023-09-06 at 9 40 12 AM

File: src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx [hunk 5]
```diff
@@ -84,8 +84,15 @@ const SwapButton = styled(StyledInternalLink)`
 export default function MobileBalanceSummaryFooter({ token }: { token: Currency }) {
   const { account } = useWeb3React()
   const balance = useCurrencyBalance(account, token)
-  const formattedBalance = formatCurrencyAmount(balance, NumberType.TokenNonTx)
-  const formattedUsdValue = formatCurrencyAmount(useStablecoinValue(balance), NumberType.FiatTokenStats)
+  const { formatCurrencyAmount } = useFormatter()
```
Comment (cbachmeier) [inline src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx:87]: Bug happens here as well
Screenshot 2023-09-06 at 9 41 27 AM

File: src/components/swap/AdvancedSwapDetails.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ import { ZERO_PERCENT } from 'constants/misc'
 import useNativeCurrency from 'lib/hooks/useNativeCurrency'
```
Comment (cbachmeier) [inline src/components/swap/AdvancedSwapDetails.tsx:9]: A few more spots in the AdvancedSwapDetails are missing currency conversion
Screenshot 2023-09-06 at 9 42 13 AM
Screenshot 2023-09-06 at 9 42 32 AM

File: src/components/swap/ConfirmSwapModal.tsx [hunk 7]
```diff
@@ -123,7 +124,10 @@ function useConfirmModalState({
   const { execute: onWrap } = useWrapCallback(
     nativeCurrency,
     trade.inputAmount.currency,
-    formatCurrencyAmount(trade.inputAmount, NumberType.SwapTradeAmount)
+    formatCurrencyAmount({
```
Comment (cbachmeier) [inline src/components/swap/ConfirmSwapModal.tsx:127]: Formatted language bug here as well
Screenshot 2023-09-06 at 9 43 40 AM

File: src/components/swap/AdvancedSwapDetails.tsx (hunk 6 above)
Comment (JackShort) [reply src/components/swap/AdvancedSwapDetails.tsx:9]: these don't use the formatCurrencyAmount function

File: src/components/AccountDrawer/MiniPortfolio/Activity/parseLocal.ts (hunk 2 above)
Comment (JackShort) [reply src/components/AccountDrawer/MiniPortfolio/Activity/parseLocal.ts:50]: a few of these use a different formatter function that I haven't gotten to converting yet

**Review by cbachmeier — APPROVED** [review]

**Review by just-toby — DISMISSED** [review]

**Review by cbachmeier — APPROVED** [review]

**Pull request merged**